nutanix/resource_nutanix_subnet.go

```diff
@@ -671,15 +717,28 @@ func getSubnetResources(d *schema.ResourceData, subnet *v3.SubnetResources) {
 		dhcpo.DomainSearchList = expandStringList(v.([]interface{}))
 	}
 
-	v, ok := d.GetOk("vlan_id")
-	if v.(int) == 0 || ok {
-		subnet.VlanID = utils.Int64Ptr(int64(v.(int)))
+	if v, ok := d.GetOk("vlan_id"); ok {
```

Comment:
why this change is required ?

Reply:
this change is required as upcoming overlay subnets will not require vlan id . it requires vpc reference .